lib/pbench/test/unit/server/query_apis/test_datasets_publish.py

```diff
@@ -105,12 +116,19 @@ def test_missing_keys(self, client, server_config, keys, user_ok, pbench_token):
         specified.
         """
         response = client.post(
-            f"{server_config.rest_uri}/datasets/publish",
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
             headers={"Authorization": "Bearer " + pbench_token},
             json=keys,
         )
         assert response.status_code == HTTPStatus.BAD_REQUEST
-        missing = [k for k in ("controller", "name", "access") if k not in keys]
+        required_keys = [
+            p[0]
+            for p in DatasetsPublish(
+                client.config, client.logger
+            ).schema.parameters.items()
+            if p[1].required
+        ]
+        missing = [k for k in required_keys if k not in keys]
         assert (
             response.json.get("message")
             == f"Missing required parameters: {','.join(missing)}"
```

Comment:
Why doesn't this call the superclass like the others?

Reply:
Ditto for other tests in the module...where these merge problems from Dave's recent commit?

Reply:
You mentioned [elsewhere](https://github.com/distributed-system-analysis/pbench/pull/2381#pullrequestreview-725880214) that some of Dave's tests weren't suitable for using the new hierarchy, but this one seems like it fits perfectly...what am I missing?

Reply:
Ah, if you mean reformatting the `required_keys`, yes I should update that in the next commit.

Reply:
Addressed in #2413.